packages/lib/hooks/useCallbackRef.ts

```diff
@@ -0,0 +1,14 @@
+import { useRef } from "react";
+
+import { useIsomorphicLayoutEffect } from "./useIsomorphicLayoutEffect";
+
+export const useCallbackRef = <C>(callback: C) => {
+  const callbackRef = useRef(callback);
+  useIsomorphicLayoutEffect(() => {
+    callbackRef.current = callback;
+  });
+
+  return callbackRef;
+};
+
+export default useCallbackRef;
```

Comment:
this really needed? I wonder if we can even just check if the last input has a value and then submit

Reply:
in order to submit we need a reference to `handleEnable`. Since, it's a function it either needs to be wrapped in a `useCallback` or a ref to get stable/most recent instance of the function.

Without the reference we could do something like finding the `form` instance in dom and calling `submit()` which isn't idiomatic in react but would simplify the code, yes.

something like:

```js
useEffect(() => {
   if (digits.at(-1)) {
      inputRef.current.form.submit()
   }
}, [digits])
```